presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java

```diff
@@ -80,7 +80,8 @@ private boolean isSupportedType(Type type)
 
         if (type instanceof ArrayType) {
             checkArgument(type.getTypeParameters().size() == 1, "expecting exactly one type parameter for array");
-            return isSupportedPrimitive(type.getTypeParameters().get(0));
+            return true;
+            //return isSupportedPrimitive(type.getTypeParameters().get(0));
```

Comment:
What did you intend to do here? Recurse to `isSupportedType(..)`?

Reply:
Yes, now it recurses.